lib/queue.js

```diff
@@ -355,6 +440,9 @@ Queue.prototype._finishJob = function (err, data, job, cb) {
   };
 
   if (err) {
+    job.options.attemptsMade += 1;
+    job.options.stacktraces.unshift(err.stack);
```

Comment:
Yeah, that doesn't matter for us - but if someone does `throw "not an error instance"` or even `throw null`, we don't want this to fail.